R/object_name_linters.R

```diff
@@ -275,32 +303,18 @@ upperalnum <- rex(one_of(upper, digit))
 alphanum   <- rex(one_of(alpha, digit))
 
 style_regexes <- list(
-  "CamelCase" = rex(start, maybe("."), upper, zero_or_more(alnum), end),
-  "camelCase" = rex(start, maybe("."), lower, zero_or_more(alnum), end),
-  "snake_case"     = rex(start, maybe("."), some_of(lower, digit), any_of("_", lower, digit), end),
-  "SNAKE_CASE"     = rex(start, maybe("."), some_of(upper, digit), any_of("_", upper, digit), end),
-  "dotted.case"    = rex(start, maybe("."), one_or_more(alphanum), zero_or_more(dot, one_or_more(alphanum)), end),
+  "symbols"     = rex(start, zero_or_more(none_of(alnum)), end),
+  "CamelCase"   = rex(start, maybe("."), upper, zero_or_more(alnum), end),
+  "camelCase"   = rex(start, maybe("."), lower, zero_or_more(alnum), end),
+  "snake_case"  = rex(start, maybe("."), some_of(lower, digit), any_of("_", lower, digit), end),
+  "SNAKE_CASE"  = rex(start, maybe("."), some_of(upper, digit), any_of("_", upper, digit), end),
+  "dotted.case" = rex(start, maybe("."), one_or_more(loweralnum), zero_or_more(dot, one_or_more(loweralnum)), end),
+  "dotted.case.mixed" = rex(start, maybe("."), one_or_more(alphanum), zero_or_more(dot, one_or_more(alphanum)), end),
```

Comment:
I think `dotted.mixed.case` is better to make the name similar to the others.

Reply:
Updated in latest commit